### PR TITLE
update entities for rollouts

### PR DESCRIPTION
**Pull request by adithyadinesh0412**
Task : Add Entity Types for Rollouts

### CODE REVIEW

File: src/database/migrations/20241122084936-add-entities-rollout.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	async up(queryInterface, Sequelize) {
+		try {
+			const defaultOrgId = queryInterface.sequelize.options.defaultOrgId
+			if (!defaultOrgId) {
+				throw new Error('Default org ID is undefined. Please make sure it is set in sequelize options.')
+			}
+
+			const entitiesArray = [
+				{
+					entityType: 'location',
```
Comment (priyanka-TL) [inline src/database/migrations/20241122084936-add-entities-rollout.js]: @adithyadinesh0412 we said we don't have a location. We need a state higher one

File: src/database/migrations/20241122084936-add-entities-rollout.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+					entities: '',
+					has_entities: true,
```
Comment (priyanka-TL) [inline src/database/migrations/20241122084936-add-entities-rollout.js:15]: @adithyadinesh0412 where is rollout table and resource table migrations. Please add those. And please create mapping tables

**Review by priyanka-TL — CHANGES_REQUESTED** [review]
Reviewed on  nov 25th  8.41 am

File: src/database/migrations/20241122084936-add-entities-rollout.js (hunk 2 above)
Comment (adithyadinesh0412) [reply src/database/migrations/20241122084936-add-entities-rollout.js:15]: That is part of another PR. I will share that. @priyanka-TL

File: src/database/migrations/20241122084936-add-entities-rollout.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+			}, [])
+
+			await queryInterface.bulkInsert('entities', entitiesFinalArray, {})
+		} catch (error) {
+			console.error('ERR : : ', error)
+		}
+	},
+
+	async down(queryInterface, Sequelize) {
+		const entityTypesList = ['location', 'role', 'gender']
```
Comment (priyanka-TL) [inline src/database/migrations/20241122084936-add-entities-rollout.js]: @adithyadinesh0412  please remove location

File: src/database/migrations/20241122084936-add-entities-rollout.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+			const entitiesArray = [
+				{
+					entityType: 'state',
+					entities: '',
+					has_entities: true,
+					is_external: true,
+					validation: {},
+				},
+				{
+					entityType: 'role',
```
Comment (priyanka-TL) [inline src/database/migrations/20241122084936-add-entities-rollout.js:20]: @adithyadinesh0412  how we know role is dependant on state ?

**Review by priyanka-TL — CHANGES_REQUESTED** [review]
Reviewed on Nov 25th 1.03 PM

File: src/database/migrations/20241122084936-add-entities-rollout.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+					entities: '',
+					has_entities: true,
+					is_external: true,
+					validation: {},
+					depended_on: 'state',
```
Comment (priyanka-TL) [inline src/database/migrations/20241122084936-add-entities-rollout.js:25]: @adithyadinesh0412  i will suggest to use id instead of value

**Review by priyanka-TL — APPROVED** [review]
approved on 25th nov, 2.49

File: src/database/migrations/20241122084739-alter-entity-type-add-isExternal-key.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
+'use strict'
+
+module.exports = {
+	up: async (queryInterface, Sequelize) => {
+		// Add the `is_external` column to the `entity_types` table
+		await queryInterface.addColumn('entity_types', 'is_external', {
```
Comment (aks30) [inline src/database/migrations/20241122084739-alter-entity-type-add-isExternal-key.js]: @adithyadinesh0412 we can create a new column called config, in which all these 3 keys will be stored as JSON

File: src/database/migrations/20241122084936-add-entities-rollout.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+				},
+				{
+					entityType: 'gender',
+					entities: [
```
Comment (aks30) [inline src/database/migrations/20241122084936-add-entities-rollout.js]: @adithyadinesh0412 this has to be external too

**Review by aks30 — CHANGES_REQUESTED** [review]
Reviewed on 28 Nov

**Pull request merged**